### PR TITLE
Fixes too high parallelism in CI

### PR DESCRIPTION
**Pull request by potiuk**
---
Issue link: WILL BE INSERTED BY [boring-cyborg](https://github.com/kaxil/boring-cyborg)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by potiuk** [comment]
Hello @ashb @Fokko @BasPH @kaxil @turbaszek @mik-laj @feluelle - I think we need to merge i quickly.  I believe I found and fixed the problem that caused the last two days constant failures of our CI jobs. 

It is very likely due to less resources (memory) available for the CI workers on Travis and I believe it was caused by too high default parallelism. In the tests we run we had default parallelism set to 32.  This means that for local executor, 32 worker processes started. I think we simply did not have enough memory to run all the processes and some workers crashed with out-of-memory exceptions and they caused the Local executor to get the EOF error on the queue reading from those processes. 

I fixed it by detecting on CI how many processes we have (2 on Travis) and setting parallelism to that value.

I got consistently failing tests without it and it started to work after that. There is one problem I have to deal with now - there is another "concurrent" test that is failing with those settings so I think I will settle on 4 rather than NUM_PROC originally.

**Review by ashb — COMMENTED** [review]
Only do this for Travis?

**Review by ashb — COMMENTED** [review]
And rather than changing the default can't we set `AIRFLOW__CORE__PARALELLISM` env var instead?

**Comment by potiuk** [comment]
Ah yeah!

**Comment by potiuk** [comment]
> And rather than changing the default can't we set `AIRFLOW__CORE__PARALELLISM` env var instead?

Done.

**Comment by potiuk** [comment]
> Only do this for Travis?

It's there we only set it when we are in CI and running tests.
```
if [[ ${CI:=} == "true" && ${RUN_TESTS} == "true" ]] ; then
```

**Review by turbaszek — APPROVED** [review]

**Comment by potiuk** [comment]
Seeems we are on a good path:

tests/cli/commands/test_task_command.py ..................               [ 44%]

File: scripts/ci/in_container/entrypoint_ci.sh [hunk 1]
```diff
@@ -193,6 +193,15 @@ fi
 # shellcheck source=scripts/ci/in_container/configure_environment.sh
 . "${MY_DIR}/configure_environment.sh"
 
+if [[ ${CI:=} == "true" && ${RUN_TESTS} == "true" ]] ; then
+    echo
+    echo " !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
+    echo "  Setting default parallellism to 4 because we can run out of memory during tests on CI"
+    echo " !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
+    echo
+    export AIRFLOW__CORE__PARALELLISM=4
```
Comment (mik-laj) [inline scripts/ci/in_container/entrypoint_ci.sh:202]: Should we set this option in config?
https://github.com/apache/airflow/blob/master/airflow/config_templates/default_test.cfg

**Comment by Fokko** [comment]
Late to the party here. I think 16 is way too high anyway. Airflow has quite a big memory footprint, so 16 airflow worker processes will consume a lot of memory. Looks good.

**Comment by potiuk** [comment]
Yep 16 is only needed by two tests - and they start MockExecutor with this high parallelism - this is just for those two tests and hopefully it gets cleaned up quickly so I hope this is not a big issue. I did not want to try to update and fix those tests requiring high parallelism - that would take a while.

File: scripts/ci/in_container/entrypoint_ci.sh (hunk 1 above)
Comment (potiuk) [reply scripts/ci/in_container/entrypoint_ci.sh:202]: I think it's better here to be only set for CI -> it's quite ok to have high parallelism for development I think.

Comment (potiuk) [reply scripts/ci/in_container/entrypoint_ci.sh:202]: Let's see if it causes future problems.

**Pull request merged**